### PR TITLE
Extend findById to accept an optional filter object

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @bajtos or @ritch

### CODE REVIEW

File: test/crud-with-options.test.js [hunk 1]
```diff
@@ -62,15 +88,49 @@ describe('crud-with-options', function () {
...
+          User.findById(u.id, filter, function(err, u) {
+            should.exist(u);
+            should.not.exist(err);
+            u.should.be.an.instanceOf(User);
+            u.should.have.property('name', 'x');
+            u.should.have.property('email', undefined);
+            done();
+          });
+        });
+      });
```
Comment (bajtos) [inline test/crud-with-options.test.js]: Can you please add a copy of this test using Promise-based API, to ensure that `findById(id, filter).then()` works correctly?

**Comment by bajtos** [comment]
I wish strong-remoting allowed mapping of query arguments to object properties, e.g. `?include=[] -> filter.include=[]`. Without that, we are getting a URL like `GET /api/cars/123?filter[include]=...`. At least it's consistent with the query parameters for `GET /api/cars`.

The implementation LGTM.

**Comment by raymondfeng** [comment]
@bajtos Added a promise-style test

**Comment by bajtos** [comment]
:+1:

**Comment by ahmed-abdulmoniem** [comment]
@bajtos @raymondfeng What about `where` in the filter object?

**Comment by j-mateo** [comment]
@ahmed-abdulmoniem This is a findByID method, short for "where "Id" equals $id". I suppose if you wanted to use the where filter, you could just use find?

**Pull request merged**